spec/api/comment_thread_spec.rb

```diff
@@ -125,6 +125,34 @@ def thread_result(params)
             rs.length.should == 0
           end
         end
+        context "when filtering own posts" do
```

Comment:
@xitij2000 Here are the tests for filtering own posts & filtering based on post type

For flagged, there are no tests added as they already exist (The functionality already existed in cs_comments_service but was simply not passed as a parameter by the discussions api), same for following